### PR TITLE
[OSD-12274] Upgrade operator-sdk to v1.21.0

### PR DESCRIPTION
**Pull request by a7vicky**
This PR migrate operator to operator-sdk version v1.21.0 and bump other dependencies

- go v1.17
- Controller Runtime  v0.11.2
- Openshift API Release v4.10 
- Kubernetes API  v0.23.5

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -44,50 +44,48 @@ func init() {
...
 
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
```
Comment (mrbarge) [inline main.go:56]: If this argument changes name, then the `Deployment` definition which supplies that argument will also need to be updated to the new name:
https://github.com/openshift/managed-node-metadata-operator/blob/c8fcec810e57c236b1fad02ef804c53ccd47846d/deploy/20_operator.yaml#L35

Comment (a7vicky) [reply main.go:56]: Thank you @mrbarge for the catch. Applied fix in my latest commit

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
Thanks for doing this.  Just a small nit.

I'm actually surprised at how much actually needed to be changed considering this operator was built with the latest version of osdk just a few months ago.

File: controllers/machineset_controller_test.go [hunk 2]
```diff
@@ -8,9 +8,8 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
-	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	machinev1 "github.com/openshift/api/machine/v1beta1"
 	m "github.com/openshift/managed-node-metadata-operator/pkg/machine"
-	corev1 "k8s.io/api/core/v1"
```
[the remark this answers is not in the document]
Comment (a7vicky) [reply controllers/machineset_controller_test.go:13]: Thanks @iamkirkbater for the review. I have addressed import alias name in my latest commit.
> I'm actually surprised at how much actually needed to be changed considering this operator was built with the latest version of osdk just a few months ago.

+1 there is nothing much change except dependecies bump. As as part of operator sdk migration epic we have planned to keep all the operator inline with minimum dependecy version of 

- Controller Runtime >=  v0.11.2
- Openshift API Release >= v4.10
- Kubernetes API >= v0.23.5

**Comment by mrbarge** [comment]
I have given this a test today.

- Verified that the operator is behaving functionally correct:
  - Confirming that changes to machinesets trigger a reconcile.
  - Confirming that adding labels to a machine's `spec.metadata.labels` are removed and synced with the machineset's labels.
- Verified that OLM upgrades the operator from 0.1.66 (current installed CSV in Production) to 0.1.90 (CSV version that results from this PR)
- Verified that metrics are available from the metrics service
- Verified that prometheus can crawl metrics from the servicemonitor without errors
- Verified leadership election managed by `cdb62f5e.managed.openshift.com` configmap is occuring successfully

All up this definitely looks good to hit staging.

**Comment by iamkirkbater** [comment]
Thanks to both of you for this!

**Pull request merged**